conftest.py

```diff
@@ -26,3 +31,37 @@ async def _load_transformers(application_config: dict) -> None:
     """Load the transformers for testing."""
     transformers = application_config.get("transformers", [])
     await init_transformers(transformers)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def _patch_redis_keys_prefix() -> None:
+    """Set the REDIS_KEYS_PREFIX for testing."""
+    settings.REDIS_KEYS_PREFIX = settings.REDIS_KEYS_PREFIX + "_test"
+
+
+@pytest.fixture(scope="session")
+async def redis_connection() -> AsyncGenerator[Redis, None]:
+    """Acquire a connection to the Redis database."""
+    redis = await init_redis()
+    yield redis
+    await redis.aclose()
+
+
+@pytest.fixture()
+async def redis(redis_connection: Redis) -> AsyncGenerator[Redis, None]:
+    """Acquire a connection to the Redis database with teardown after each test."""
+    yield redis_connection
+
+    # Delete any existing keys created during the test
+    keys_to_delete = [
+        key async for key in redis_connection.scan_iter(f"{settings.REDIS_KEYS_PREFIX}:*")
+    ]
+
+    if keys_to_delete:
+        await redis_connection.delete(*keys_to_delete)
```

Comment:
@thijsfranck , `redis_connection.delete` should be awaited. And you should use `@pytest_asyncio.fixture`